concepts/patterns/introduction.md

```diff
@@ -0,0 +1,75 @@
+# Introduction
+
+Recall from the Fundamentals concept, an AWK program is composed of **pattern-action pairs**.
+
+```awk
+pattern1 { action1 }
+pattern2 { action2 }
+...
+```
+
+## What do we mean by "pattern"?
+
+The "pattern" is any AWK expression that results in a true or false value.
+
+## The Empty Pattern
+
+The pattern can be omitted.
+In this case, the action is performed for every record.
+
+We can print all the usernames in the passwd file.
+
+```sh
+awk -F: '{print $1}' /etc/passwd
+```
+
+## Regular Expressions
+
+AWK can do what grep can do.
+
+This example compares the whole record (`$0`) to the regex.
+
+```sh
+awk '/regex/' data.txt
+```
+
+A regular expression literal is enclosed in `/` slashes.
+
+Use the `~` regex-matching operator to match a particular field. 
```

Comment:
The ~ isn't specific to matching particular fields. You could do `var_a ~ var_b`. 

It might be better to lead with explaining that ~ is the regex operator ... and there is a default operation, such that `/foo/ ~ $0`  can be written as `/foo/`